### PR TITLE
buttons - use children instead of props

### PR DESCRIPTION
**Pull request by emma-floored**
@Jinsoo94 @MagdalenaHenke let me know how you feel about this! I restructured the button component and got rid of the `ButtonPrimary`/`ButtonSecondary`/`ButtonDanger` components.

### CODE REVIEW

**Comment by Jinsoo94** [comment]
great add to delete ratio! looks good to me. now that `isDisabled` -> `disabled` has happened here, should we aim to apply it everywhere during the sprint?

**Comment by emma-floored** [comment]
@Jinsoo94 yeah, I think that would be a good idea (`isDisabled` -> `disabled`)

**Comment by Jinsoo94** [comment]
yikes the conflicts

**Comment by MagdalenaHenke** [comment]
checking this out now!

File: packages/blocks-base/styles/buttons.styl [hunk 1]
```diff
@@ -58,9 +58,13 @@ $button-danger-active-text-color = $secondary-00;
   max-width: 480px;
   min-width: 44px;
   margin: 0;
-  padding: 8px 18px; // specified padding - 2px to account for border
+  padding: 10px 18px; // specified padding - 2px to account for border
```
[the remark this answers is not in the document]
Comment (emma-floored) [reply packages/blocks-base/styles/buttons.styl]: oh the buttons always have a 2px border (it's just transparent when not focused - the reason to do that is so the button doesn't move when you add a 2px focus border), so I subtract that size from the desired padding so the button still looks the same size as the spec

File: packages/blocks-docs/src/code_examples/ButtonIconExample.js [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
-    />
-    <ButtonSecondary
-      onClick={() => {}}
-      icon={<MoreIcon />}
-    />
-    <ButtonDanger
-      onClick={() => {}}
-      icon={<UpIcon />}
-    />
+    <Button color="primary" type="icon" onClick={() => {}}>
```
[the remark this answers is not in the document]
Comment (emma-floored) [reply packages/blocks-docs/src/code_examples/ButtonIconExample.js]: I can see how that might be nice... however I'm not totally sure how I would want to export that information, yet, so I'm going to leave it for a possible follow up!

**Pull request merged**